src/main/java/com/aws/greengrass/detector/IpDetectorService.java

```diff
@@ -61,10 +72,38 @@ public void startup() throws InterruptedException {
      * @throws  InterruptedException if the thread interrupted
      */
     @Override
-    public void shutdown() throws InterruptedException {
+    public synchronized void shutdown() throws InterruptedException {
```

Comment:
No, I am not calling this method. I saw even super methods are not synchronized. I think we should keep this so that we can avoid any unforeseen problem. Let me know your thoughts

Reply:
I agree with Jason, Component lifecycle should ensure non-concurrency here.